### PR TITLE
Add jwks field validation to avoid sidecar crash

### LINKED ISSUE #33053: Creating a RequestAuthentication object with an invalid 'jwks' parameter causes istio-proxy sidecars to break
**Bug description**

I run the following test deployment / service exposed via an ingress gateway:

<details>
  <summary>Deployment / service / gateway / virtualservice definitions</summary>

```yaml
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: my-app
  labels:
    app: my-app
spec:
  replicas: 1
  selector:
    matchLabels:
      app: my-app
  template:
    metadata:
      labels:
        app: my-app
    spec:
      containers:
      - name: main
        image: kennethreitz/httpbin
        ports:
        - containerPort: 80
---
apiVersion: v1
kind: Service
metadata:
  name: my-app-service
spec:
  selector:
    app: my-app
  ports:
    - protocol: TCP
      port: 80
      targetPort: 80


---
# Ingress GW
apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  name: my-ingress-gateway
spec:
  selector:
    istio: ingressgateway # use Istio default gateway implementation
  servers:
  - port:
      number: 80
      name: http
      protocol: HTTP
    hosts:
    - "httpbin.myhost.com"
---
# Routes mapping for the GW
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: my-httpbin-virtual-service
spec:
  hosts:
  - "httpbin.myhost.com"
  gateways:
  - my-ingress-gateway
  http:
  - match:
    - uri:
        prefix: /httpbin
    route:
    - destination:
        port:
          number: 80
        host: my-app-service
    rewrite:
      uri: /
```
</details>

This works just fine. At this point, **I added by mistake a RequestAuthentication with an invalid `jwks`**:

```yaml
apiVersion: security.istio.io/v1beta1
kind: RequestAuthentication
metadata:
  name: my-service-req-authn
spec:
  selector:
    matchLabels:
      app: my-app
  jwtRules:
  - issuer: christophe
    jwks: foo # <-- this is invalid
```

As soon as this is created, it completely breaks istio-proxy for the pods of this workload. Here's the output of istio-proxy:

```
2021-05-24T14:55:15.870513Z	warn	Envoy proxy is NOT ready: config not received from Pilot (is Pilot running?): cds updates: 1 successful, 0 rejected; lds updates: 0 successful, 1 rejected
2021-05-24T14:55:17.870519Z	warn	Envoy proxy is NOT ready: config not received from Pilot (is Pilot running?): cds updates: 1 successful, 0 rejected; lds updates: 0 successful, 1 rejected
```

As soon as I fix the RequestAuthentication object by a valid one, e.g.

```yaml
apiVersion: security.istio.io/v1beta1
kind: RequestAuthentication
metadata:
  name: my-service-req-authn
spec:
  selector:
    matchLabels:
      app: my-app
  jwtRules:
  - issuer: christophe
    jwks: >
      { "keys":[ {"e":"AQAB","kid":"DHFbpoIUqrY8t2zpA2qXfCmr5VO5ZEr4RzHU_-envvQ","kty":"RSA","n":"xAE7eB6qugXyCAG3yhh7pkDkT65pHymX-P7KfIupjf59vsdo91bSP9C8H07pSAGQO1MV_xFj9VswgsCg4R6otmg5PV2He95lZdHtOcU5DXIg_pbhLdKXbi66GlVeK6ABZOUW3WYtnNHD-91gVuoeJT_DwtGGcp4ignkgXfkiEm4sw-4sfb4qdt5oLbyVpmW6x9cfa7vs2WTfURiCrBoUqgBo_-4WTiULmmHSGZHOjzwa8WtrtOQGsAFjIbno85jp6MnGGGZPYZbDAa_b3y5u-YpW7ypZrvD8BgtKVjgtQgZhLAGezMt0ua3DRrWnKqTZ0BJ_EyxOGuHJrLsn00fnMQ"}]}
 
```

... then the sidecars start working again:

```
2021-05-24T14:56:55.870391Z	warn	Envoy proxy is NOT ready: config not received from Pilot (is Pilot running?): cds updates: 1 successful, 0 rejected; lds updates: 0 successful, 1 rejected
2021-05-24T14:56:57.870620Z	warn	Envoy proxy is NOT ready: config not received from Pilot (is Pilot running?): cds updates: 1 successful, 0 rejected; lds updates: 0 successful, 1 rejected
2021-05-24T14:56:59.870984Z	info	Initialization took 9m41.838941787s
2021-05-24T14:56:59.871013Z	info	Envoy proxy is ready
```

Here's the output of `istioctl bug-report` when it's in the state `Envoy proxy is NOT ready`:

[bug-report.tar.gz](https://github.com/istio/istio/files/6533236/bug-report.tar.gz)

[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[x] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Expected behavior** The `RequestAuthentication` resource is refused at creation time or ignored

**Steps to reproduce the bug**
- Deploy the deployment / service / gateway / virtualservice provided above
- Deploy the (invalid) RequestAuthentication object
- Kill a pod to have a new one started by the deployment controller
- Notice how the sidecar is not working

**Version** (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version --short` if you used Helm)

```
$ istioctl version --remote
client version: 1.10.0
control plane version: 1.10.0
data plane version: 1.10.0 (3 proxies)

$ kubectl version --short
Client Version: v1.21.1
Server Version: v1.20.2
```

**How was Istio installed?** Using `istioctl install --verify`

**Environment where the bug was observed (cloud vendor, OS, etc)** DigitalOcean managed Kubernetes service

### PR DESCRIPTION
**Pull request by kebe7jun**
Fix #33053

[X] Security

[X] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Review by irisdingbj — COMMENTED** [review]
lgtm to me. @yangminzhu PTAL

**Comment by kebe7jun** [comment]
@yangminzhu would you please take a look here?

**Review by hzxuzhonghu — APPROVED** [review]

File: pkg/config/validation/validation.go [hunk 1]
```diff
@@ -1779,6 +1780,13 @@ func validateJwtRule(rule *security_beta.JWTRule) (errs error) {
 		}
 	}
 
+	if rule.Jwks != "" {
+		_, err := jwt.Parse([]byte(rule.Jwks))
+		if err != nil {
+			errs = multierror.Append(errs, errors.New("jwks parse error"))
```
Comment (yangminzhu) [inline pkg/config/validation/validation.go:1786]: nit: better to include the actual error in the output.

**Pull request merged**